components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIAdminImpl.java

```diff
@@ -538,8 +538,12 @@ public KeyManagerConfigurationDTO addKeyManagerConfiguration(
                                 APIUtil.getInternalOrganizationDomain(keyManagerConfigurationDTO.getOrganization()));
                 keyManagerConfigurationDTO.setExternalReferenceId(identityProvider.getResourceId());
             } catch (IdentityProviderManagementException e) {
+                ExceptionCodes exceptionCode = ExceptionCodes.IDP_ADDING_FAILED;
+                if (e.getMessage().contains("has already been registered")) {
+                    exceptionCode = ExceptionCodes.DUPLICATE_ISSUER;
```

Comment:
rather check for exception message can't we check for existence

Reply:
To check the issuer, we have to get all the idps registered in the org one by one and check the org right? Do we have any other way. 
As I checked in apimgt db we do not keep the issuer in a separate field.